sympy/sets/fancysets.py

```diff
@@ -659,9 +659,13 @@ def _contains(self, other):
                 _ = self.size  # validate
             except ValueError:
                 return
-        ref = self.start if self.start.is_finite else self.stop
-        if (ref - other) % self.step:  # off sequence
-            return S.false
+        if self.start.is_finite:
```

Comment:
nit: we already know if `ref - other` is going to be finite by looking at self.stop:
```python
if self.start.is_finite:
    ref = self.start
elif self.stop.is_finite:
    ref = self.stop
else:
    return other.is_Integer
if (ref - other) % self.step:
    return S.false
return _sympify(other >= self.inf and other <= self.sup)
```